www/includes/easyparliament/templates/html/interests/category.php

```diff
@@ -0,0 +1,137 @@
+
+<div class="full-page static-page toc-page">
+    <div class="full-page__row">
+
+        <div class="toc-page__col">
+        <?php if ($selected_category_id) { ?>
+        <div class="toc js-toc">
+            <ul>
+            <li><a href="?>&chamber=<?= $chamber_slug ?>"><?= gettext("Back to categories list")?></a></li>
+
+                <?php foreach ($categories as $category_id => $category_name) { ?>
+                    <li><a href="?chamber=<?= $chamber_slug ?>&category_id=<?= $category_id ?>"><?= $category_name ?></a></li>
+                <?php }; ?>
+            </ul>
+        </div>
+        <?php }; ?>
+
+        </div>
+        <div class="toc-page__col">
+
+            <div class="panel">
+                <h1>📒<?= gettext('Register of interests')?></h1>
+                <h2><?= $register->displayChamber() ?> - <?= $register->published_date ?></h2>
+                <p><?= gettext('This page shows the latest version of the register of interests by category.') ?></p>
+                <?php if (LANGUAGE == 'cy') { ?>
+                    <p><?= gettext('For more information, see the official Senedd page') ?></a>.
+                <?php } else { ?>
+                    <p>For more information on the different categories, see the <a href="<?= $register->officialUrl() ?>">the official <?= $register->displayChamber() ?> page</a>.
+                <?php } ?>
+                <p><a href="/interests/"><?= gettext('Read more about our registers data')?></a>.</p>
+                <hr>
+                <?php /** @var MySociety\TheyWorkForYou\DataClass\Regmem\Register $register */ ?>
+
+                <?php if ($selected_category_id === null) { ?>
+                    <h3><?= gettext('Categories') ?></h3>
+                    <ul>
+                        <?php foreach ($categories as $category_id => $category_name) { ?>
+                            <li><a href="?chamber=<?= $chamber_slug ?>&category_id=<?= $category_id ?>"><?= $category_name ?></a></li>
+                        <?php }; ?>
+                    </ul>
+                <?php }; ?>
+
+
+                <?php foreach ($categories as $category_id => $category_name) { ?>
+                    <?php if ($selected_category_id != $category_id) {
+                        continue;
+                    }; ?>
```

Comment:
would it not make more sense to just set a `$selected_category_name` at the data stage and pass that in to the template and then you could avoid this loop?

Reply:
Yep! Done a change on this